x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpResource.java

```diff
@@ -143,30 +140,36 @@ private synchronized boolean blockUntilCheckAndPublish(final RestClient client)
      * still be dirty at the end, but the success of it will still return based on the checks it ran.
      *
      * @param client The REST client to make the request(s).
-     * @return {@code true} if the resource is available for use. {@code false} to stop.
+     * @param listener Returns {@code true} if the resource is available for use. {@code false} to stop.
      * @see #isDirty()
      */
-    public final synchronized boolean checkAndPublish(final RestClient client) {
+    public final void checkAndPublish(final RestClient client, final ActionListener<Boolean> listener) {
         // we always check when asked, regardless of clean or dirty
         state.set(State.CHECKING);
 
-        boolean success = false;
+        doCheckAndPublish(client, new ActionListener<Boolean>() {
```

Comment:
maybe use `ActionListener.wrap(...)`:

```
doCheckAndPublish(client, ActionListener.wrap(success -> {
            state.compareAndSet(State.CHECKING, success ? State.CLEAN : State.DIRTY);
            listener.onResponse(success);
        }, e -> {
            state.compareAndSet(State.CHECKING, State.DIRTY);
            listener.onFailure(e);
        }));
```

I think using that reads easier than an anonymous inner class.